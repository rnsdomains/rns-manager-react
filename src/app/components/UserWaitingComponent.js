import React from 'react';
import propTypes from 'prop-types';
import { Spinner } from 'react-bootstrap';

const UserWaitingComponent = ({ message, visible }) => {
  if (!visible) {
    return (<></>);
  }

  return (
    <div className="row waiting">
      <div className="col-md-6 offset-md-3">
        <Spinner animation="grow" variant="primary" />
        <p>{message}</p>
      </div>
    </div>
  );
};

UserWaitingComponent.defaultProps = {
  message: '',
  visible: true,
};

UserWaitingComponent.propTypes = {
  message: propTypes.string,
  visible: propTypes.bool,
};

export default UserWaitingComponent;
