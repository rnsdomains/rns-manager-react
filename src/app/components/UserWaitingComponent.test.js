import React from 'react';
import { mount } from 'enzyme';

import UserWaitingComponent from './UserWaitingComponent';

describe('UserWaitingComponent', () => {
  it('renders and matches snapshot', () => {
    const component = mount(<UserWaitingComponent message="Please Wait!" />);
    expect(component.find('p').text()).toBe('Please Wait!');
    expect(component).toMatchSnapshot();
  });
});
