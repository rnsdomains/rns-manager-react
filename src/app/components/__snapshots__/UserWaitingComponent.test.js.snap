// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserWaitingComponent renders and matches snapshot 1`] = `
<UserWaitingComponent
  message="Please Wait!"
  visible={true}
>
  <div
    className="row waiting"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <Spinner
        animation="grow"
        variant="primary"
      >
        <div
          className="spinner-grow text-primary"
        />
      </Spinner>
      <p>
        Please Wait!
      </p>
    </div>
  </div>
</UserWaitingComponent>
`;
